test/test_websocket_grid.py

```diff
@@ -0,0 +1,320 @@
+from app.websocket.app import create_app
+from flask import Flask, session, request, json as flask_json
+from flask_socketio import (
+    SocketIO,
+    send,
+    emit,
+    join_room,
+    leave_room,
+    Namespace,
+    disconnect,
+)
+
+import unittest
+import threading
+import multiprocessing
+import grid as gr
+import syft as sy
+import torch as th
+import time
+import requests
+import json
+
+hook = sy.TorchHook(th)
+
+
+gateway_url = "http://localhost:8080"
+
+ports = ["3000", "3001", "3002"]
+ids = ["Bob", "Alice", "James"]
+
+
+def setUpGateway(port):
+    import os
+
+    os.environ["SECRET_KEY"] = "Secretkeyhere"
+    from gateway.app import create_app
+
+    app = create_app(debug=False)
+    app.run(host="0.0.0.0", port="8080")
+
+
+def setUpNodes(config):
+    from app.websocket.app import create_app, socketio
+
+    requests.post(
+        gateway_url + "/join",
+        data=json.dumps(
+            {
+                "node-id": config[0],
+                "node-address": "http://localhost:" + config[1] + "/",
+            }
+        ),
+    )
+    app = create_app(debug=False)
+    socketio.run(app, host="0.0.0.0", port=port)
+
+
+# Init Grid Gateway
+p = multiprocessing.Process(target=setUpGateway, args=("8080",))
+p.daemon = True
+p.start()
+time.sleep(5)
+
+# Init Grid Nodes
+for (node_id, port) in zip(ids, ports):
+    config = (node_id, port)
+    p = multiprocessing.Process(target=setUpNodes, args=(config,))
+    p.daemon = True
+    p.start()
+
+time.sleep(20)
+
```

Comment:
Everything before this line should be at conftest.py

Reply:
Done! (fe59983)